koku/api/report/queries.py

```diff
@@ -680,9 +680,12 @@ def _get_tag_group_by(self):
         group_by = []
         tag_groups = self.get_tag_group_by_keys()
         for tag in tag_groups:
-            sanitized_tag = sanitize_tag(strip_prefix(tag, TAG_PREFIX))
+            original_tag = strip_prefix(tag, TAG_PREFIX)
+            sanitized_tag = sanitize_tag(original_tag)
             tag_db_name = self._mapper.tag_column + "__" + sanitized_tag
-            group_pos = self.parameters.url_data.index(sanitized_tag)
+            encoded_tag = str.encode(original_tag)
+            encoded_tag_url = quote_from_bytes(encoded_tag, safe=URL_ENCODED_SAFE)
```

Comment:
Since `original_tag` is already a string, there's no reason to convert to bytes before encoding.
```suggestion
            encoded_tag_url = quote(encoded_tag, safe=URL_ENCODED_SAFE)
```